avocado/utils/software_manager.py

```diff
@@ -517,6 +557,63 @@ def provides(self, name):
         else:
             return None
 
+    def build_dep(self, name):
+        """
+        Install build-dependencies for package [name]
+
+        :param name: name of the package
+
+        :return True: If build dependencies are installed properly
+        """
+
+        try:
+            process.system('yum-builddep -y --tolerant %s' % name, sudo=True)
+            return True
+        except process.CmdError, details:
+            log.error(details)
+            return False
+
+    def get_source(self, name, dest_path):
+        """
+        Downloads the source package and prepares it in the given dest_path
+        to be ready to build.
+
+        :param name: name of the package
+        :param dest_path: destination_path
+
+        :return final_dir: path of ready-to-build directory
+        """
+        path = tempfile.mkdtemp(prefix='avocado_software_manager')
+        if dest_path is None:
+            log.error("Please provide a valid path")
+            return ""
+        for pkg in ["rpm-build", "yum-utils"]:
+            if not self.check_installed(pkg):
+                if not self.install(pkg):
+                    log.error("SoftwareManager (YumBackend) can't get packages"
+                              "with dependency resolution: Package '%s'"
+                              "could not be installed", pkg)
+                    return ""
+        try:
+            src_rpm = process.system_output('yumdownloader --urls --source'
+                                            ' %s' % name).splitlines()[-1]
+            src_rpm = src_rpm.split("/")[-1]
+            process.system_output('yumdownloader --source %s --destdir %s' % (name, path))
```

Comment:
it's unlikely but between the first and second invocation cache might be refreshed and package of a different version might be downloaded. Anyway merging it as is and we can improve later (verified the yum/dnf backend and looked at the rest).